server/src/main/java/io/crate/rest/action/HttpError.java

```diff
@@ -151,6 +152,8 @@ public static HttpError fromThrowable(Throwable throwable) {
             httpErrorStatus = HttpErrorStatus.SNAPSHOT_WITH_SAME_NAME_EXISTS_ALREADY;
         } else if (throwable instanceof UnavailableShardsException) {
             httpErrorStatus = HttpErrorStatus.ONE_OR_MORE_SHARDS_NOT_AVAILABLE;
+        } else if (throwable instanceof DocumentMissingException) {
+            httpErrorStatus = HttpErrorStatus.DOCUMENT_NOT_FOUND;
```

Comment:
Newer pattern is to implement `httpErrorStatus` on the exception if it's a `ElasticsearchException`.
We should probably add a comment about that within this method. The same is also the case for the PG error code.

Reply:
Looks like as a follow up PR we could also remove `RestStatus status` it's not used in a meaningful way and overlaps with the more detailed `HttpErrorStatus`

Reply:
Ah thanks, will change.